crates/sui-core/src/authority_active/gossip/mod.rs

```diff
@@ -75,11 +78,26 @@ pub async fn gossip_process_with_start_seq<A>(
     let mut gossip_tasks = FuturesUnordered::new();
 
     loop {
+        if _active_authority.state.committee.load().epoch != committee.epoch {
+            // If epoch has changed, we need to make a new copy of the active authority,
+            // and update all local variables.
+            // We also need to remove any authority that's no longer a valid validator
+            // from the list of peer names.
+            // It's ok to keep the existing gossip tasks running even for peers that are no longer
+            // validators, and let them end naturally.
```

Comment:
I think we need to update structures in the `health` field of active authority too?

Reply:
What would be the consequence to not touch the health?

Reply:
At an epoch change, active participation is required for the authorities to remain as active authorities right? Some authorities will be removed, so we can reset the health since at that time our current authorities were just elected so are presumed to be healthy. We also no longer need to track health for the ones that left.